src/frontend/src/utils/userNumber.test.ts

```diff
@@ -0,0 +1,81 @@
+import { getUserNumber, setUserNumber } from "./userNumber";
+
+testLocalStorage("user number can be set", () => {
+  expect(getUserNumber()).toBe(undefined);
+  setUserNumber(BigInt(123456));
+  expect(getUserNumber()).toBe(BigInt(123456));
+});
+
+testLocalStorage("user number can be cleared", () => {
+  setUserNumber(BigInt(123456));
+  expect(getUserNumber()).toBe(BigInt(123456));
+  setUserNumber(undefined);
+  expect(getUserNumber()).toBe(undefined);
+});
+
+testLocalStorage(
+  "user number localStorage format",
+  () => {
+    setUserNumber(BigInt(123456));
+  },
+  { after: { userNumber: "123456" } }
+);
+
+testLocalStorage(
+  "user number read from known localStorage format",
+  () => {
+    expect(getUserNumber()).toStrictEqual(BigInt(123456));
+  },
+  { before: { userNumber: 123456 } }
+);
+
+/** Run a test that makes use of localStorage. Local storage is always cleared after
+ * the test was run.
+ * If `before` is specified, local storage is populated with its content before the test is run.
+ * If `after` is specified, the content of local storage are checked against `after` after the
+ * test is run and before local storage is cleared.
+ */
+function testLocalStorage(
+  name: string,
+  fn: () => void,
+  opts?: { before?: LocalStorage; after?: LocalStorage }
+) {
+  test(name, () => {
+    localStorage.clear();
+    if (opts?.before !== undefined) {
+      setLocalStorage(opts.before);
+    }
+    fn();
+    if (opts?.after !== undefined) {
+      const actual: LocalStorage = readLocalStorage();
+      const expected: LocalStorage = opts.after;
+      expect(actual).toStrictEqual(expected);
+    }
+
+    localStorage.clear();
+  });
+}
+
+/// Type representing the whole localStorage, used for tests.
+type LocalStorage = Record<string, unknown>;
+
+/// Set the entire localStorage (does NOT clear localStorage)
+function setLocalStorage(ls: LocalStorage) {
+  for (const key in ls) {
+    localStorage.setItem(key, JSON.stringify(ls[key]));
+  }
+}
+
+function readLocalStorage(): LocalStorage {
+  let ls: LocalStorage = {};
```

Comment:
was curious if it can be inlined, so just in case

```
Object.entries(localStorage).map(([key, value]) => ({[key]: value}))
```

Reply:
Also, doesn't this map to an array of `{key: value}`s, i.e. `[{key1: value1}, {key2: value2}, ...]`?

Reply:
either way, I'm curious now :) but it's less clear, so I'll probably stick to the current solution

Reply:
> The Object.entries() method returns an array of a given object's own enumerable string-keyed property [key, value] pairs. [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)

`getItem` and `setItem` are no properties but functions.

> doesn't this map to an array

You are right. It would need a `reduce`

```
Object.entries(localStorage).reduce((acc, [key, value]) => ({...acc, [key]: value}), {})
```

Seems to work but yes current solution is fine, was curiosity too from my side.

<img width="1536" alt="Capture d’écran 2022-10-25 à 10 29 03" src="https://user-images.githubusercontent.com/16886711/197724347-cd93c179-93ee-4f00-99d3-4773d020da84.png">